From 64ad16f2c6c957b22468ecfa205fc94d2450c893 Mon Sep 17 00:00:00 2001
From: Tyrel Datwyler <tyreld@linux.ibm.com>
Date: Mon, 15 Mar 2021 15:07:20 -0700
Subject: [PATCH] ibmvfc: disable MQ channelization by default

References: bsc#1184570 ltc#192356
Patch-mainline: never, temporary default change

The MQ feature at this time is showing instability. Disable it by
default.

Signed-off-by: Tyrel Datwyler <tyreld@linux.ibm.com>
Acked-by: Michal Suchanek <msuchanek@suse.de>
---
 drivers/scsi/ibmvscsi/ibmvfc.c | 2 +-
 drivers/scsi/ibmvscsi/ibmvfc.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/ibmvscsi/ibmvfc.c b/drivers/scsi/ibmvscsi/ibmvfc.c
index a1a7132aae09..ef0058289c9e 100644
--- a/drivers/scsi/ibmvscsi/ibmvfc.c
+++ b/drivers/scsi/ibmvscsi/ibmvfc.c
@@ -57,7 +57,7 @@ MODULE_LICENSE("GPL");
 MODULE_VERSION(IBMVFC_DRIVER_VERSION);
 
 module_param_named(mq, mq_enabled, uint, S_IRUGO);
-MODULE_PARM_DESC(mq, "Enable multiqueue support. "
+MODULE_PARM_DESC(mq, "Enable (EXPERIMENTAL) multiqueue support. "
 		 "[Default=" __stringify(IBMVFC_MQ) "]");
 module_param_named(scsi_host_queues, nr_scsi_hw_queues, uint, S_IRUGO);
 MODULE_PARM_DESC(scsi_host_queues, "Number of SCSI Host submission queues. "
diff --git a/drivers/scsi/ibmvscsi/ibmvfc.h b/drivers/scsi/ibmvscsi/ibmvfc.h
index 19dcec3ae9ba..dd09a0f122e5 100644
--- a/drivers/scsi/ibmvscsi/ibmvfc.h
+++ b/drivers/scsi/ibmvscsi/ibmvfc.h
@@ -41,7 +41,7 @@
 #define IBMVFC_DEFAULT_LOG_LEVEL	2
 #define IBMVFC_MAX_CDB_LEN		16
 #define IBMVFC_CLS3_ERROR		0
-#define IBMVFC_MQ			1
+#define IBMVFC_MQ			0
 #define IBMVFC_SCSI_CHANNELS		8
 #define IBMVFC_MAX_SCSI_QUEUES		16
 #define IBMVFC_SCSI_HW_QUEUES		8
-- 
2.26.2

